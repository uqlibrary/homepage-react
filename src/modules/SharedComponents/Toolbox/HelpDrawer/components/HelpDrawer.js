import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Drawer from '@mui/material/Drawer';
import Button from '@mui/material/Button';
import withStyles from '@mui/styles/withStyles';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

/* istanbul ignore next */
const styles = theme => ({
    paper: {
        width: 320,
        paddingTop: 40,
        paddingRight: 40,
        paddingLeft: 40,
        paddingBottom: 0,
        maxHeight: '100%',
    },
    title: {
        color: theme.palette.primary.main,
    },
    button: {
        float: 'right',
        marginBottom: 60,
    },
});

export class HelpDrawer extends Component {
    static propTypes = {
        open: PropTypes.bool.isRequired,
        title: PropTypes.string.isRequired,
        text: PropTypes.any.isRequired,
        hide: PropTypes.func.isRequired,
        buttonLabel: PropTypes.string,
        classes: PropTypes.object,
    };
    static defaultProps = {
        buttonLabel: 'CLOSE',
    };

    render() {
        const { classes, title, text, buttonLabel, open, hide } = this.props;
        let indexedText = null;
        if (this.props.text && this.props.text.props && this.props.text.props.children) {
            indexedText = React.Children.map(this.props.text.props.children, (child, index) => {
                if (child.type) {
                    return React.cloneElement(child, { key: index });
                } else {
                    return child;
                }
            });
        }
        return (
            <Drawer classes={{ paper: classes.paper }} open={open} anchor="right" onClose={hide}>
                <Grid container spacing={5} id="help-drawer">
                    <Grid item xs={12}>
                        <Typography
                            key={'title'}
                            component={'span'}
                            variant={'h6'}
                            className={classes.title}
                            id="help-drawer-title"
                        >
                            {title}
                        </Typography>
                        <Typography
                            key={'text'}
                            component={'span'}
                            variant="body2"
                            id={`help-drawer-text-${title.replace(/\s/g, '')}`}
                        >
                            {indexedText || text}
                        </Typography>
                    </Grid>
                    <Grid item xs={12} id="help-drawer-button">
                        <Button
                            variant="contained"
                            color="primary"
                            className={classes.button}
                            children={buttonLabel}
                            onClick={hide}
                        />
                    </Grid>
                </Grid>
            </Drawer>
        );
    }
}

export default withStyles(styles, { withTheme: true })(HelpDrawer);
