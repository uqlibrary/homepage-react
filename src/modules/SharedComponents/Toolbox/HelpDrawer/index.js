export { default as helpDrawerReducer } from './reducer';
