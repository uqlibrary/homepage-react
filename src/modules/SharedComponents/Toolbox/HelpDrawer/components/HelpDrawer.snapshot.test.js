import React from 'react';
import { HelpDrawer } from './HelpDrawer';
import HelpDrawerWithStyles from './HelpDrawer';

function setup(testProps = {}) {
    const props = {
        classes: {},
        theme: { palette: { white: { main: '#FFFFFF' } } },
        open: true,
        title: 'Test title',
        text: 'Test text',
        hide: jest.fn(),
        buttonLabel: 'Test OK',
        ...testProps,
    };
    return getElement(HelpDrawer, props);
}

describe('HelpDrawer snapshots tests', () => {
    it('renders menu', () => {
        const hdText = 'Integer mattis rutrum velit nec posuere. Quisque rhoncus quam elit.';
        const wrapper = setup({ title: 'HelpDrawer Title', text: hdText, open: true, buttonLabel: 'Close' });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders text as react children', () => {
        const wrapper = setup({
            title: 'HelpDrawer title',
            text: (
                <p>
                    <span>Test text</span>
                </p>
            ),
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders text as react element', () => {
        const wrapper = setup({
            title: 'HelpDrawer title',
            text: <span>Test text</span>,
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with styles', () => {
        const wrapper = getElement(HelpDrawerWithStyles, {
            open: true,
            title: 'Test title',
            text: 'Test text',
            hide: jest.fn(),
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
