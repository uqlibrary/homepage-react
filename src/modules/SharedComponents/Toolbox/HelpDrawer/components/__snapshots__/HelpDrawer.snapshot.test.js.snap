// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpDrawer tests renders text as react element 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`HelpDrawer tests should render with styles 1`] = `
<div
  aria-hidden="true"
/>
`;
